### PR TITLE
feat(fireblocks-sdk): bring your own auth provider

### PR DESCRIPTION
**Pull request by kennethtxytqw**
Our team would like to use our own auth provider like a custom AWS-KMS agent to sign our keys instead.

### CODE REVIEW

**Comment by kennethtxytqw** [comment]
@tomervil Can you take a look at this please?

File: src/fireblocks-sdk.ts [hunk 1]
```diff
@@ -50,8 +50,8 @@ export class FireblocksSDK {
      * @param apiKey Your api key. This is a uuid you received from Fireblocks
      * @param apiBaseUrl The fireblocks server URL. Leave empty to use the default server
      */
-    constructor(privateKey: string, apiKey: string, apiBaseUrl: string = "https://api.fireblocks.io") {
-        this.authProvider = new ApiTokenProvider(privateKey, apiKey);
+    constructor(privateKey: string, apiKey: string, apiBaseUrl: string = "https://api.fireblocks.io", authProvider: IAuthProvider = undefined) {
+        this.authProvider = authProvider === undefined ? new ApiTokenProvider(privateKey, apiKey): authProvider;
```
Comment (tomervil) [inline src/fireblocks-sdk.ts:54]: @kennethtxytqw Sorry for the late reply. Do you mind rewriting it as
`this.authProvider = authProvider ?? new ApiTokenProvider(privateKey, apiKey);`?
I'd be happy to merge

Comment (kennethtxytqw) [reply src/fireblocks-sdk.ts]: Ah good catch!

Comment (kennethtxytqw) [reply src/fireblocks-sdk.ts]: Fixed!

**Pull request merged**